extensions/raycast-monkeytype-theme/src/raycast-monkeytype-theme.tsx

```diff
@@ -0,0 +1,285 @@
+    themeJson["colors"]["green"] = "#07BA65";
+    themeJson["colors"]["blue"] = "#0A7FF5";
+    themeJson["colors"]["purple"] = "#470AF5";
+    themeJson["colors"]["magenta"] = "#F50AA3";
+
+    return JSON.stringify(themeJson, null, 2);
+  }
+
+  if (isLoading) {
+    return <List isLoading={isLoading} searchBarPlaceholder="Loading themes..." />;
+  }
+
+  async function validateIdentity(themeName: string) {
+    if (!environment.canAccess(AI)) {
+      const options: Alert.Options = {
+        icon: { source: getImageUrl(themeName) },
+        title: "Install " + themeName + " Theme",
+        message: "Custom Themes require an active Pro subscription",
+        primaryAction: {
+          title: "Upgrade to Pro",
+          onAction: () => {
+            open("https://www.raycast.com/pro");
+          },
+        },
+      };
+      await confirmAlert(options);
+      return false;
+    }
+    return true;
+  }
+
+  return (
+    <List
+      isShowingDetail
+      searchBarPlaceholder="Filter by theme name"
+      // searchBarAccessory={
+      //   <List.Dropdown tooltip="Filter by Appearance" onChange={setAppearanceFilter}>
+      //     <List.Dropdown.Item title="All" value="all" icon={Icon.BulletPoints} />
+      //     <List.Dropdown.Item title="Light" value="light" icon={Icon.Sun} />
+      //     <List.Dropdown.Item title="Dark" value="dark" icon={Icon.Moon} />
+      //   </List.Dropdown>
+      // }
+    >
+      {themes.map((theme) => (
+        <List.Item
+          key={theme.name}
+          title={theme.name}
+          keywords={[theme.appearance]}
+          icon={{
+            source: getImageUrl(theme.name),
+          }}
+          // accessories={[
+          //   { icon: getIconColor(theme.colors.background), tooltip: "Background Color" },
+          //   { icon: getIconColor(theme.colors.text), tooltip: "Text Color" },
+          //   { icon: getIconColor(theme.colors.selection), tooltip: "Selection Color" },
+          //   { icon: getIconColor(theme.colors.loader), tooltip: "Loader Color" },
+          // ]}
+          actions={
+            <ActionPanel>
+              <Action
+                title="Add Light Appearance to Raycast"
+                icon={Icon.Sun}
+                onAction={async () => {
+                  if (!(await validateIdentity(theme.name))) return;
+                  open(theme["ray.so.add.light.url"]);
+                  closeMainWindow();
+                }}
+                shortcut={{ modifiers: ["cmd"], key: "l" }}
+              />
+              <Action
+                title="Add Dark Appearance to Raycast"
+                icon={Icon.Moon}
+                onAction={async () => {
+                  if (!(await validateIdentity(theme.name))) return;
+                  open(theme["ray.so.add.dark.url"]);
+                  closeMainWindow();
+                }}
+                shortcut={{ modifiers: ["cmd"], key: "d" }}
+              />
```

Comment:
Since we have `appearance` in the data received, wouldn't it be better to just show either light or dark

Reply:
The appearance in data is determined by the background color, but this alone doesn't always reflect the best theme. Some themes are designed to look better with a light appearance, even if they are dark. Therefore, I provide both light and dark options for users to choose the one that suits them best.